src/liballoc/slice.rs

```diff
@@ -1722,6 +1722,54 @@ impl<T> [T] {
         // NB see hack module in this file
         hack::into_vec(self)
     }
+
+    /// Creates a [`Vec`] by repeating a slice `n` times.
+    ///
+    /// [`Vec`]: ../../std/vec/struct.Vec.html
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// assert_eq!([1, 2].repeat(3), vec![1, 2, 1, 2, 1, 2]);
+    /// ```
+    #[stable(feature = "repeat_slice", since = "1.27.0")]
+    pub fn repeat(&self, n: usize) -> Vec<T> where T: Copy {
+        if n == 0 {
+            return Vec::new();
+        }
+
+        // If `n` is larger than zero, it can be split as
+        // `n = 2^expn + rem (2^expn > rem, expn >= 0, rem >= 0)`.
+        // `2^expn` is the number represented by the leftmost '1' bit of `n`,
+        // and `rem` is the remaining part of `n`.
+
+        // Using `Vec` to access `set_len()`.
+        let mut buf = Vec::with_capacity(self.len() * n);
+
+        // `2^expn` repetition is done by doubling `buf` `expn`-times.
+        buf.extend(self);
+
```

Comment:
You have missed this part: https://github.com/sinkuu/rust/blob/3d58543d49266a7ec3eb5f5f2ffaf902fce17c53/src/liballoc/str.rs#L2084-L2102